### PR TITLE
Add GaussMarkovProcess pybind11 interface and examples

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

This PR creates a GaussMarkovProcess Pybind11 interface and adds examples

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: examples/gauss_markov_process_example.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# You can plot the data generated by this program by following these
+# steps.
+#
+# 1. Run this program and save the output to a file:
+#     ./gauss_markov_process > plot.data
```
Comment (chapulina) [inline examples/gauss_markov_process_example.py:19]: I think we should tell the user to set `PYTHONPATH` like we do on other examples:

https://github.com/ignitionrobotics/ign-math/blob/1cac6547f141e5f5872d8d2976ebc1ac50633135/examples/angle_example.py#L18-L21

```suggestion
#     python3 gauss_markov_process_example.py > plot.data
```

File: examples/gauss_markov_process_example.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# You can plot the data generated by this program by following these
+# steps.
+#
+# 1. Run this program and save the output to a file:
+#     ./gauss_markov_process > plot.data
+#
+# 2. Use gnuplot to create a plot:
+#     gnuplot -e 'set terminal jpeg; plot "plot.data" with lines' > out.jpg
+require 'math'
```
Comment (chapulina) [inline examples/gauss_markov_process_example.rb:23]:
```suggestion
require 'ignition/math'
```

File: examples/gauss_markov_process_example.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# You can plot the data generated by this program by following these
+# steps.
+#
+# 1. Run this program and save the output to a file:
+#     ./gauss_markov_process > plot.data
```
Comment (chapulina) [inline examples/gauss_markov_process_example.rb:19]: Here too we should tell the user to set the env var:

https://github.com/ignitionrobotics/ign-math/blob/1cac6547f141e5f5872d8d2976ebc1ac50633135/examples/angle_example.rb#L18-L21

And

```suggestion
#     ruby gauss_markov_process.rb > plot.data
```

**Review by chapulina — CHANGES_REQUESTED** [review]
Just some comments about the examples

**Review by chapulina — APPROVED** [review]
LGTM, I pushed some tweaks to the examples in b3e1a977691646c046878389445b47279ef77127

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**